### PR TITLE
CLN: Unreachable code, Boolean comparison, duplicate functions

### PR DESCRIPTION
**Pull request by mroeschke**
Breaking up #21974.

This batch is:

- Unreachable code
- Redefined function (in asv)
- Boolean equality comparison

### CODE REVIEW

**Comment by jbrockmendel** [comment]
LGTM.  

@jorisvandenbossche IIRC you wanted to keep numpydoc changes separate from others?

File: doc/sphinxext/numpydoc/docscrape_sphinx.py [hunk 1]
```diff
@@ -56,10 +56,6 @@ def _str_indent(self, doc, indent=4):
 
     def _str_signature(self):
         return ['']
-        if self['Signature']:
-            return ['``%s``' % self['Signature']] + ['']
-        else:
-            return ['']
```
Comment (gfyoung) [inline doc/sphinxext/numpydoc/docscrape_sphinx.py]: Yikes!  That's pretty bad...

**Review by gfyoung — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
> IIRC you wanted to keep numpydoc changes separate from others?

Yes, numpydoc is vendored. So if we want to make changes to it, it should be done upstream.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**